Updated docs and release notes for licensed community (#3184)
FusionAuth/fusionauth-site: astro/src/content/docs/release-notes/index.mdx

Looking for release notes older than 1.23.0? Look in the [release notes archive](/docs/release-notes/archive). Looking to be [notified of new releases?](/docs/operate/roadmap/releases#release-notifications) <span class="not-prose no-underline"><a class="ml-2" href="/docs/releases.xml"><i class="fas fa-xs fa-rss text-orange-700 text-2xl" width="50px" /></a></span>

<ReleaseNoteHeading version='1.52.0' releaseDate='August xx, 2024' />

<GeneralMigrationWarning>

>   When designing integrations that leverage the User Registrations API, please be aware that the `data` field for the FusionAuth application, Id `3c219e58-ed0e-4b18-ad48-f4f92793ae32`, may now contain a `preferences` object with variable fields within.

Author: Could be more generic here, stating that the `data` field for the FusionAuth app is protected and may include a variety of different fields.

Reviewer: I'm confused. Is it the `data` field on the application? Or the `registration.data` field for a user who is registered for the FusionAuth admin UI?

I'm confused because you mention the `User Registrations API`.

Author: It is the `registration.data` field for a user who is registered for the FusionAuth admin UI

Reviewer: How about this, then?

```suggestion
  When using the User Registrations API, the `data` field for the FusionAuth application with Id `3c219e58-ed0e-4b18-ad48-f4f92793ae32` may now contain a `preferences` object. This object is reserved and should not be modified.
```

  Removal of this field is targeted for the end of 2024.
</DeprecationWarning>

### New

> * Version `1.52.0` introduces support for Licensed Community. After upgrading, those with a community plan can now get even more out of FusionAuth with a free license.

Author: Feels like we perhaps should have a link to a blog post here for users to get more information on Licensed Community.

Reviewer: Better to have 'more information' as part of the docs. I don't think we push folks from docs -> blog except in the narrow 'related blog posts' case.
